examples/jsm/nodes/core/InputNode.js

```diff
@@ -1,80 +1,113 @@
+import { Color, Matrix3, Matrix4, Vector2, Vector3, Vector4 } from 'three';
 import Node from './Node.js';
 
-class InputNode extends Node {
+function getValueType( value ) {
 
-	constructor( inputType ) {
+	if ( typeof value === 'number' ) {
 
-		super( inputType );
+		return 'float';
 
-		this.inputType = inputType;
+	} else if ( typeof value === 'boolean' ) {
 
-		this.constant = false;
+		return 'bool';
 
-	}
+	} else if ( value?.isVector2 === true ) {
 
-	setConst( value ) {
+		return 'vec2';
 
-		this.constant = value;
+	} else if ( value?.isVector3 === true ) {
 
-		return this;
+		return 'vec3';
 
-	}
+	} else if ( value?.isVector4 === true ) {
 
-	getConst() {
+		return 'vec4';
 
-		return this.constant;
+	} else if ( value?.isMatrix3 === true ) {
 
-	}
+		return 'mat3';
 
-	getInputType( /* builder */ ) {
+	} else if ( value?.isMatrix4 === true ) {
 
-		return this.inputType;
+		return 'mat4';
+
+	} else if ( value?.isColor === true ) {
+
+		return 'color';
 
 	}
 
-	getInputHash( builder ) {
+}
 
-		return this.getHash( builder );
+function getValueFromType( type ) {
 
-	}
+	if ( type === 'color' ) {
 
-	generateConst( builder ) {
+		return new Color();
 
-		return builder.getConst( this.getNodeType( builder ), this.value );
+	} else if ( type.includes( 'vec2' ) ) {
 
-	}
+		return new Vector2();
 
-	generate( builder, output ) {
+	} else if ( type.includes( 'vec3' ) ) {
 
-		const type = this.getNodeType( builder );
+		return new Vector3();
 
-		if ( this.constant === true ) {
+	} else if ( type.includes( 'vec4' ) ) {
 
-			return builder.format( this.generateConst( builder ), type, output );
+		return new Vector4();
 
-		} else {
+	} else if ( type.includes( 'mat3' ) ) {
 
-			const inputHash = this.getInputHash( builder );
+		return new Matrix3();
 
-			let sharedNode = builder.getNodeFromHash( inputHash );
+	} else if ( type.includes( 'mat4' ) ) {
 
-			if ( sharedNode === undefined ) {
+		return new Matrix4();
 
-				builder.setHashNode( this, inputHash );
+	}
 
-				sharedNode = this;
+}
 
-			}
+class InputNode extends Node {
 
-			const inputType = sharedNode.getInputType( builder );
+	constructor( value, nodeType ) {
 
-			const nodeUniform = builder.getUniformFromNode( sharedNode, builder.shaderStage, inputType );
-			const propertyName = builder.getPropertyName( nodeUniform );
+		if ( nodeType === undefined ) {
 
-			return builder.format( propertyName, type, output );
+			nodeType = getValueType( value );
```

Comment:
We must add this in `.getNodeType()`. Constructor fields must always be "passive" in the sense that we can later `set` a `value` after create the class, like:

```js
const myVec3 = new Nodes.UniformNode();
myVec3.value = new THREE.Vector3();
```

Reply:
Done in a slightly different way.